sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -283,89 +524,118 @@ ssize_t gcoap_fileserver_handler(coap_pkt_t *pdu, uint8_t *buf, size_t len, void
         strncpy(request.namebuf, entry->root, sizeof(request.namebuf));
         namelength = strlen(entry->root);
     }
-
     bool is_directory = true; /* either no path component at all or trailing '/' */
-    coap_optpos_t opt = {
-        .offset = coap_get_total_hdr_len(pdu),
-    };
-    uint8_t *value;
-    ssize_t optlen;
-    while ((optlen = coap_opt_get_next(pdu, &opt, &value, 0)) != -ENOENT) {
-
-        if (optlen < 0) {
-            errorcode = COAP_CODE_BAD_REQUEST;
-            goto error;
+    {
```

Comment:
Why the new scope?

Reply:
to limit the scope of inside defined variables. Do you not think that this is worth the scope overhead?